src/main/java/io/cryostat/jfr/datasource/server/JfrResource.java

```diff
@@ -126,6 +129,35 @@ void list(RoutingContext context) {
         response.end(responseBuilder.toString());
     }
 
+    @Route(path = "/delete_all", methods = HttpMethod.DELETE)
+    @Blocking
+    void deleteAll(RoutingContext context) {
+        HttpServerResponse response = context.response();
+        setHeaders(response);
+
+        final StringBuilder stringBuilder = new StringBuilder();
+        List<String> deletedFiles = deleteAllFiles();
+
+        for (String deletedFile : deletedFiles) {
+            stringBuilder.append("Deleted: " + deletedFile);
+            stringBuilder.append(System.lineSeparator());
+        }
+
+        response.end(stringBuilder.toString());
+    }
+
+    @Route(path = "/delete", methods = HttpMethod.DELETE)
+    @Blocking
+    void delete(RoutingContext context) {
+        HttpServerResponse response = context.response();
+        setHeaders(response);
+
+        String fileName = context.getBodyAsString();
+        deleteFile(fileName);
```

Comment:
What if the `fileName` doesn't exist? I think that should be a `404` response.

Reply:
You could do `filename.strip()` on the request body parameter, but whatever behaviour the existing endpoints have is how these new deletion endpoints should behave as well. If the others include trailing whitespaces in the expected file name to load for example, then this should do the same.